test/state/state.cpp

```diff
@@ -181,6 +182,68 @@ std::variant<TransactionReceipt, std::error_code> transition(
         return acc.destructed || (rev >= EVMC_SPURIOUS_DRAGON && acc.erasable && acc.is_empty());
     });
 
-    return TransactionReceipt{gas_used, host.take_logs()};
+    auto receipt =
+        TransactionReceipt{result.status_code == EVMC_SUCCESS, gas_used, host.take_logs()};
+
+    // Cannot put it into constructor call because logs are std::moved from host instance.
+    receipt.logs_bloom_filter = compute_bloom_filter(receipt.logs);
+
+    return receipt;
+}
+
+namespace
+{
+inline bytes legacy_tx_to_rlp(uint64_t nonce, const intx::uint256& gas_price, uint64_t gas_limit,
```

Comment:
```suggestion
bytes legacy_tx_to_rlp(uint64_t nonce, const intx::uint256& gas_price, uint64_t gas_limit,
```
Don't put `inline` if not sure it make sense. Also these helper functions are one-liners so not sure they make some much sense at this point.